delta/subcommands/classify.py

```diff
@@ -121,6 +121,10 @@ def __init__(self, loss_object):
 
     def update_state(self, y_true, y_pred, sample_weight=None): #pylint: disable=unused-argument, arguments-differ
         this_loss = self._loss_object.call(y_true, y_pred)
+        # In some situations the loss will not be a single number, take the mean to make it one
+        this_loss = tf.cond(tf.math.equal(tf.rank(this_loss), 0),
```

Comment:
The loss should not always be reduced to a single number. It is valid to have multiple dimensions. Sometimes it is important to preserve that even when reducing, if there are different batch sizes they will be waited differently in the reducer.

Reply:
Do our metric classes in metrics.py also need to change?  They use reduce_add in a similar manner.

Reply:
I had misunderstood what I was looking at, sorry! This is the right thing to do!